raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -918,22 +948,21 @@ private FetchQuorumRecordsResponseData tryCompleteFetchQuorumRecordsRequest(
         long fetchOffset = request.fetchOffset();
         int lastFetchedEpoch = request.lastFetchedEpoch();
         LeaderState state = quorum.leaderStateOrThrow();
-        Optional<LogOffsetMetadata> highWatermark = state.highWatermark();
```

Comment:
This was another minor issue. We can return the high watermark after the update is applied to improve latency on the follower in some cases.